src/main/java/com/gzs/main/NoConnectionException.java

```diff
@@ -0,0 +1,10 @@
+package com.gzs.main;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class NoConnectionException extends RuntimeException {
+    public NoConnectionException(Throwable error) {
```

Comment:
@DespotOfMorea better call this "cause" instead of "error" (as errors and exceptions are different things and cause will more likely be exception, not error).